### PR TITLE
Remove precision_plugin pre_dispatch() method

### LINKED ISSUE #10884: Remove precision_plugin pre_dispatch() method
## Proposed refactor

There is no logic in pre_dispatch() in any precision_plugins. We can remove the 

### Motivation

Simplify the logic
Eventually we should merge pre_dispatch and dispatch for strategies too. The first step is to remove 

### Pitch

Remove 
https://github.com/PyTorchLightning/pytorch-lightning/blob/038c151b6ee0754adb66714a62bf32b0a389511f/pytorch_lightning/plugins/precision/precision_plugin.py#L204-L205

And references
https://github.com/PyTorchLightning/pytorch-lightning/blob/1d2878523ac36703b851f0646b0ef17f14c582cc/pytorch_lightning/accelerators/accelerator.py#L82

### Additional context

part of #10648 
______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @justusschock @awaelchli @akihironitta @tchaton

### PR DESCRIPTION
**Pull request by four4fish**
## What does this PR do?
There is no logic in pre_dispatch() in any precision_plugins. 

Simplify the logic
Eventually we should merge pre_dispatch and dispatch for strategies too. The first step is to remove

With this change #10885 could be simplified

Fixes #10884 
## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/precision/precision_plugin.py [hunk 1]
```diff
@@ -201,9 +201,6 @@ def clip_grad_by_norm(self, optimizer: Optimizer, clip_val: Union[int, float]) -
         parameters = self.main_params(optimizer)
         torch.nn.utils.clip_grad_norm_(parameters, clip_val)
 
-    def pre_dispatch(self) -> None:
```
Comment (four4fish) [inline pytorch_lightning/plugins/precision/precision_plugin.py:204]: @reviewers Accelerator refactor and stable version will have a lot of breaking changes. I think customized precision_plugin + overrides pre_dispatch() use case should be very minor. 
I can change this to deprecated if that is preferred

**Review by awaelchli — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
